### DISCUSSION TITLE
SAP oData services

### DISCUSSION THREAD

**Discussion by nrw303** [odata2ts/odata2ts#325] [General]
I'm an ABAP developer learning web dev and typescript... does anyone have an example of a simple app that uses an SAP on prem oData v2 service.
I think I have the setup all good, I just do see how to call the damn thing!

help a newbie out with some examples please :)

**Comment by texttechne** [comment]
Hi @nrw303,

regarding odata2ts, the main documentation should be enough, I guess: [Main Service](https://odata2ts.github.io/docs/odata-client/the-main-service)...

Regarding examples in combination with UI5, see this example repo: [ui5 with odata2ts](https://github.com/odata2ts/playground/tree/main/packages/ui5-odata-model-interaction)

**Reply by texttechne** [reply]
If that doesn't help, give some specifics: Are you using UI5? Where are you stuck (even pseudo code would help)?

And maybe last but not least: Are we talking about odata2ts at all or UI5's ODataModel?

**Comment by nrw303** [comment]
thanks for replying, much appreciated. I went through the Main Service docs and these were super helpful, shame on me for glossing over that page previously, it's like the most important page in all the docs! I also republished my web api to oData v4 (so much better now!) together with those and a bunch of exploration got me sorted.

BTW I'm not using UI5, I'm using Svelte 5 with Kit, it's all working nicely now. Thanks for the great library!

**Reply by texttechne** [reply]
Yeah, I need to overhaul the documentation. Problem there is, that there are different use cases. Some people (usually UI5 users, although there's also a good argument to use odata2ts within UI5...) only require the models and not the full OData client. There is also a potential use case to not use the full client but also q-objects and the query builder, but that is not a prominent use case and got too much attention in the docs... So no blame on you, it just points to the fact, that the documentation can be improved.

I would always recommend V4, it's much more powerful and concise, except for one scenario: Using UI5 with SAP's ODataModel. My colleagues had painful experiences there...

Last but not least, could you mark my first reply as answer in this discussion. Then we can close it out.

P.S: Kudos for diving so deep into frontend stuff and using Svelte!

**Reply by nrw303** [reply]
I don't see an option to mark as an answer? I googled and found [this thread](https://github.com/orgs/community/discussions/3892) bit that doesn't really help as even though your first answer is the answer, I don't see a way to mark it as such.

anyway, yeah I'm loving Svelte, esp the v5 syntax is really cool. and thanks once again for a cool and helpful library.
